src/devices/nm-device-bridge.c

```diff
@@ -275,16 +276,32 @@ update_connection (NMDevice *device, NMConnection *connection)
 	NMSettingBridge *s_bridge = nm_connection_get_setting_bridge (connection);
 	int ifindex = nm_device_get_ifindex (device);
 	const Option *option;
+	gs_free char *stp = NULL;
+	int stp_value;
 
 	if (!s_bridge) {
 		s_bridge = (NMSettingBridge *) nm_setting_bridge_new ();
 		nm_connection_add_setting (connection, (NMSetting *) s_bridge);
 	}
 
-	for (option = master_options; option->name; option++) {
+	option = master_options;
+	nm_assert (nm_streq (option->name, "stp_state"));
+
+	stp = nm_platform_sysctl_master_get_option (nm_device_get_platform (device), ifindex, option->sysname);
+	nm_assert (stp);
+
+	stp_value = strtol (stp, NULL, 10);
```

Comment:
don't use `strtol()`. It is locale dependent. Also, you lack error checking here. Use `_nm_utils_ascii_str_to_int64()`.

Yes, that is also below. Please add error checking there too.

Also, fix calling `g_object_set()` without checking that the integer value is in a suitable range. First of all, various of these properties are `g_param_spec_uint()` so passing `int value` is at least ugly (if not wrong). If you fix the uint vs. int issue, you could use `nm_g_object_set()`, which returns an error instead of asserting when the value is out of range.

Reply:
We can use _nm_utils_ascii_str_to_int64() to do also the check on the nm allowed range: if out of that, the default nm value will be assumed.
Ok for value as a uint.